src/geometries/primitives/parametrizedcurve.jl

```diff
@@ -0,0 +1,42 @@
+# ------------------------------------------------------------------
+# Licensed under the MIT License. See LICENSE in the project root.
+# ------------------------------------------------------------------
+
+"""
+    ParametrizedCurve(func, range = (0.0, 1.0))
+
+A parametrized curve is a curve defined by a function `func` that maps a parameter `t` to a `Point` in space.
+The parameter `t` is defined in the interval `[a, b]`. The curve can only be evaluated for `t` in the
+interval `[0, 1]`.
+
+## Examples
+
+```julia
+ParametrizedCurve(t -> Point(cospi(2t), sinpi(2t)))
+```
+"""
+struct ParametrizedCurve{M<:Manifold,C<:CRS,F<:Function,R<:Tuple} <: Primitive{M,C}
+  func::F
+  range::R
+
+  function ParametrizedCurve(func, range=(0.0, 1.0))
+    a = first(range)
+    p = func(a)
```

Comment:
I wonder if we should pass the range as an actual Julia range `a:b` to avoid situations where the `typeof(a) != typeof(b)` and where the inferred `typeof(p)` is not adequate to store all points along the range.

Alternatively, we need to `promote` the passed range values to a common type before inferring `p`.

Reply:
A range `a:b` is a `StepRangeLen`, i.e. it has a step length (defaulting to 1.0, i.e. `0:2pi` becomes `0.0:1.0:6.0` for instance), which seems inappropriate here. I made a proposal in [fefc71e](https://github.com/JuliaGeometry/Meshes.jl/pull/1093/commits/fefc71e55930aa4804eac41e67f7bc9dce016096) to `promote` the range values.